version: '3.8'

name: ft_transcendence

services:
  django:
    build: ./django
    container_name: django
    restart: always
    env_file: ./.env
    volumes:
      - type: bind
        source: ./django
        target: /app
    networks:
      - network
    depends_on:
      db:
        condition: service_healthy

  db:
    image: postgres
    container_name: django-db
    restart: always
    env_file: ./.env
    volumes:
      - db:/var/lib/postgresql/data
    networks:
      - network
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U $$POSTGRES_USER"]
      interval: 5s
      timeout: 5s
      retries: 5

  nginx:
    build: ./nginx
    container_name: nginx
    ports:
      - '1443:443'
    networks:
      - network
    depends_on:
      - django
    volumes:
      - ./nginx/nginx.conf:/etc/nginx/nginx.conf

volumes:
  db:

networks:
  network:
    driver: bridge
